engine/crates/engine-v2/schema/src/walkers/mod.rs

```diff
@@ -52,22 +52,15 @@ impl<'a, Id: Copy> SchemaWalker<'a, Id>
 where
     Schema: std::ops::Index<Id>,
 {
-    pub fn get(&self) -> &'a <Schema as std::ops::Index<Id>>::Output {
-        &self.schema[self.wrapped]
+    // Clippy complains because it's ambiguous with AsRef. But AsRef doesn't allow us to add the 'a
+    // lifetime. I could rename to `to_ref()` or `ref()`, but doesn't feel better than `as_ref()`.
+    #[allow(clippy::should_implement_trait)]
```

Comment:
There's examples of `as_ref()` methods without trait impl even in the std library (https://doc.rust-lang.org/std/option/enum.Option.html#method.as_ref) so I agree this is the right name here, clippy is being unnecessarily pedantic.